features/frameworks/mbed-client-cli/source/ns_cmdline.c

```diff
@@ -113,35 +202,64 @@
 #define DEFAULT_PROMPT "/>"
 #define VAR_PROMPT "PS1"
 #define VAR_RETFMT "RETFMT"
+#define MBED_CMDLINE_ESCAPE_BUFFER_SIZE 10
 
+// by default use
+#ifndef MBED_CONF_CMDLINE_ENABLE_ALIASES
+#define MBED_CONF_CMDLINE_ENABLE_ALIASES 1
+#endif
+#ifndef MBED_CONF_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS
+#define MBED_CONF_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS 0
+#elif defined(MBED_CONF_CMDLINE_ENABLE_INTERNAL_COMMANDS) && MBED_CONF_CMDLINE_ENABLE_INTERNAL_COMMANDS == 1 && MBED_CONF_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS == 1
+#warning "Cannot set MBED_CONF_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS along with MBED_CONF_CMDLINE_ENABLE_INTERNAL_COMMANDS"
+#endif
 // Maximum length of input line
-#ifdef MBED_CMDLINE_MAX_LINE_LENGTH
-#define MAX_LINE MBED_CMDLINE_MAX_LINE_LENGTH
-#else
-#define MAX_LINE 2000
+#ifndef MBED_CONF_CMDLINE_MAX_LINE_LENGTH
+#define MBED_CONF_CMDLINE_MAX_LINE_LENGTH 2000
```

Comment:
Typo? This is a ridiculously large default buffer size and the `mbed_lib.json` states this:

https://github.com/artokin/mbed-os/blob/9325f17d7066856b6504fd0d27e21d85de307f0e/features/frameworks/mbed-client-cli/mbed_lib.json#L44-L47

"defaults to **200** when not set"

Reply:
The lib-file is actually incorrect in this case. That is valid value for complex automation tests.